src/Faker/Provider/es_MX/Person.php

```diff
@@ -0,0 +1,337 @@
+            $firstName = self::removeAccents(mb_strtoupper($firstName?$firstName: static::firstNameMale()));
+        } else {
+            $firstName = self::removeAccents(mb_strtoupper($firstName?$firstName: static::firstNameFemale()));
+        }
+
+        $lastNameFather = self::removeAccents(mb_strtoupper($lastNameFather?$lastNameFather: static::randomElement(static::$lastNames)));
+        $lastNameMother = self::removeAccents(mb_strtoupper($lastNameMother?$lastNameMother: static::randomElement(static::$lastNames)));
+        $birthDate = $birthDate?$birthDate:\Faker\Provider\DateTime::dateTimeBetween();
+
+        $rfc = self::commonPart('rfc', $firstName, $lastNameFather, $lastNameMother, $birthDate);
+        $rfc .= strtoupper(static::lexify('???'));
+
+        return $rfc;
+    }
+
+    /**
+     * HELPERS
+     */
+
+    protected static function commonPart($type, $firstName = null, $lastNameFather = null, $lastNameMother = null, $birthDate = null)
+    {
+        $common  = substr($lastNameFather, 0, 1);
+        $common .= self::firstInternalVowel($lastNameFather);
+        $common .= $lastNameMother?substr($lastNameMother, 0, 1):'X';
+        $common .= substr($firstName, 0, 1);
+        if ($type == 'curp') {
+            $common = str_replace(array_keys(self::$badWordsCURP), array_values(self::$badWordsCURP), $common);
+        } else {
+            $common = str_replace(array_keys(self::$badWordsRFC), array_values(self::$badWordsRFC), $common);
+        }
+        $common .=$birthDate->format('ymd');
+        return $common;
+    }
+
+    protected static function firstInternalVowel($word)
+    {
+        if (preg_match('([AEIOU])', $word, $matches)) {
+            return $matches[0];
+        }
+        return 'X';
+    }
+
+    protected static function firstInternalConsonant($word)
+    {
+        if (preg_match('([BCDFGHJKLMNPQRSTUVWXYZ])', $word, $matches)) {
+            return $matches[0];
+        }
+        return 'X';
+    }
+
+    protected static function removeCommonNames($name)
+    {
+        foreach (self::$notAcceptedNames as $notAcceptedName) {
+            $name = preg_replace($notAcceptedName, '', $name);
+        }
+        return $name;
+    }
+    protected static function removePrefixes($name)
+    {
+        foreach (self::$prefixes as $prefix) {
+            $name = preg_replace($prefix, '', $name);
+        }
+        return $name;
+    }
+    protected static function removeAccents($name)
+    {
+        return str_replace(array_keys(self::$accents), array_values(self::$accents), $name);
+    }
+
+    protected static $notAcceptedNames = array(
+        '/^MARIA DEL /',
+        '/^MARIA DE LOS /',
+        '/^MARIA /',
+        '/^JOSE DE /',
+        '/^JOSE /',
+        '/^MA. /',
+        '/^MA /',
+        '/^M. /',
+        '/^J. /',
+        '/^J /'
+    );
+
+    protected static $prefixes = array(
+        '/^DE /',
+        '/^DEL /'
+    );
+
+    protected static $badWordsCURP = array(
```

Comment:
https://preescolarveracruz.wordpress.com/2011/08/19/aviso-urgente/catalogo-de-palabras-inconvenientes-curp/ basically this are swearing words in mexico so they are not supposed to be used as part of the curp